### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by RayRedGoose**
**What's this PR do?**
This adds styles and the functionality to the LoginPage component such as:
 - LoginForm component to control entered data
 - LoginInput and LoginRadioButton components to add name, favorite quote and rank
 - ErrorNotification if any input is empty when submit button is clicked

Testing contains:
 - snapshots for every components
 - change state methods in LoginForm component
 - submit action for correct and error path in LoginForm component
 - invoke change state method passed as prop in LoginInput and LoginRadioButton components

NOTE: all files were moved to specific login folder to organize components, that's why changes include some deletes 

**Where should the reviewer start?**
The better way to make review is LoginPage -> LoginForm -> LoginInput/LoginRadioButton -> ErrorNotification, from parent to children. All important changes is located at LoginForm component.

**How should this be manually tested?**
You can pull down and run the tests, or we can just do it in person during work time.

### CODE REVIEW

File: src/components/login/ErrorNotification/ErrorNotification.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import './ErrorNotification.scss'
+
+const ErrorNotification = () => {
+  return (
```
Comment (spaceplesiosaur) [inline src/components/login/ErrorNotification/ErrorNotification.js:5]: You made new components for error notifications?  What was the motivation for this decision?

File: src/components/login/LoginForm/LoginForm.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        this.setState({data: {...this.state.data, name: value}, error: false})
+        break;
+      case 'favorite quote':
+        this.setState({data: {...this.state.data, quote: value}, error: false})
+        break;
+      default:
+        this.setState({data: {...this.state.data, rank: value}, error: false})
+        break;
+    }
+  }
```
Comment (spaceplesiosaur) [inline src/components/login/LoginForm/LoginForm.js:32]: Can you talk me through the above function when you get a chance?  I'm unfamiliar with the syntax :-D

File: src/components/login/LoginForm/LoginForm.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+  submitUser = (event) => {
+    event.preventDefault()
+    if (!this.checkError()) {
+      this.props.addUser({...this.state.data})
```
Comment (spaceplesiosaur) [inline src/components/login/LoginForm/LoginForm.js:37]: addUser is part of App.js?  You're probably ok to add that file to this PR now that my stuff is in there.

File: src/components/login/LoginForm/LoginForm.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      this.clearInputs()
+    } else {
+      this.setState({...this.state, error: true})
+    }
+  }
+
+  checkError = () => {
+    const valName = this.state.data.name.replace(' ', '') === ''
+    const valQuote = this.state.data.quote.replace(' ', '') === ''
```
Comment (spaceplesiosaur) [inline src/components/login/LoginForm/LoginForm.js:46]: Just went over this - this is distinguishing between white space and an actual blank field - good thought!

File: src/components/login/LoginForm/LoginForm.scss [hunk 5]
```diff
@@ -0,0 +1,31 @@
+$font-dark: #5A2C05;
+$border-color: #F1AA61;
```
Comment (spaceplesiosaur) [inline src/components/login/LoginForm/LoginForm.scss:2]: I didn't use variable, though I know I used the same colors - ok with making a variable file at some point?

File: src/components/login/LoginForm/LoginForm.test.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+      expect(form.state('data').name).toEqual('')
+
+      form.instance().toggleChangeState('name', 'Yoda')
+
+      expect(form.state('data').name).toEqual('Yoda')
+    })
+
+    it('should update quote state when toggleChangeState is called with first argument as a favorite quote', () => {
+
```
Comment (spaceplesiosaur) [inline src/components/login/LoginForm/LoginForm.test.js:35]: Love how you tested all of the possible state changes here!

**Review by spaceplesiosaur — APPROVED** [review]
Can you send me the methods on App.js?

File: src/components/login/ErrorNotification/ErrorNotification.js (hunk 1 above)
Comment (RayRedGoose) [reply src/components/login/ErrorNotification/ErrorNotification.js:5]: I made new component for error notification, login inputs and login radio button for clean up my render in login form.

File: src/components/login/LoginForm/LoginForm.js (hunk 2 above)
Comment (RayRedGoose) [reply src/components/login/LoginForm/LoginForm.js:32]: Switch is equivalent of multiple if statement. `switch (type)` sets up what we want to check. When we have any value for this, it looking for case with the same value. `case 'name'` is equal to `if(type === 'name')`, before `break` there is code that should be complete for this case, `break` stops all switch and doesn't invoke cases laying below.

File: src/components/login/LoginForm/LoginForm.js (hunk 3 above)
Comment (RayRedGoose) [reply src/components/login/LoginForm/LoginForm.js:37]: yes, addUser is prop function, I'll add it later.

File: src/components/login/LoginForm/LoginForm.scss (hunk 5 above)
Comment (RayRedGoose) [reply src/components/login/LoginForm/LoginForm.scss:2]: I plan to add vars in different file on css refactoring stage.

File: src/components/login/LoginForm/LoginForm.test.js (hunk 6 above)
Comment (RayRedGoose) [reply src/components/login/LoginForm/LoginForm.test.js:35]: thank you!

**Pull request merged**